homeassistant/components/squeezebox/media_player.py

```diff
@@ -141,38 +152,21 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     hass.data[DATA_SQUEEZEBOX].extend(media_players)
     async_add_entities(media_players)
 
-    async def async_service_handler(service):
-        """Map services to methods on MediaPlayerEntity."""
-        method = SERVICE_TO_METHOD.get(service.service)
-        if not method:
-            return
+    platform = entity_platform.current_platform.get()
 
-        params = {
-            key: value for key, value in service.data.items() if key != "entity_id"
-        }
-        entity_ids = service.data.get("entity_id")
-        if entity_ids:
-            target_players = [
-                player
-                for player in hass.data[DATA_SQUEEZEBOX]
-                if player.entity_id in entity_ids
-            ]
-        else:
-            target_players = hass.data[DATA_SQUEEZEBOX]
+    platform.async_register_entity_service(
+        SERVICE_CALL_METHOD, SQUEEZEBOX_CALL_METHOD_SCHEMA, "async_call_method",
```

Comment:
If we pass a dict with validation inside instead of a voluptuous schema, the entity service helper will make an entity schema out of the dict for us.

https://github.com/home-assistant/core/blob/26241980d76781104e267da7255fc5ef1345021b/homeassistant/helpers/config_validation.py#L790-L806

See the example here:
https://developers.home-assistant.io/docs/dev_101_services#entity-services